packages/libs/coreui/src/components/inputs/SelectList.tsx

```diff
@@ -30,31 +37,36 @@ export default function SelectList<T extends string>({
   isDisabled = false,
   isLoading = false,
   instantUpdate = false,
+  transitionDuration,
   ...props
 }: SelectListProps<T>) {
   const [selected, setSelected] = useState<SelectListProps<T>['value']>(value);
   const [buttonDisplayContent, setButtonDisplayContent] = useState<ReactNode>(
     getDisplayContent(value, items, defaultButtonDisplayContent)
   );
+  const [isOpen, setIsOpen] = useState(false);
 
-  const onClose = () => {
-    onChange(selected);
+  const onClose = useCallback(() => {
+    if (!instantUpdate) onChange(selected);
     setButtonDisplayContent(
       getDisplayContent(selected, items, defaultButtonDisplayContent)
     );
-  };
+  }, [instantUpdate, selected, items, defaultButtonDisplayContent, onChange]);
 
   /**
    * Keep caller up to date with any selection changes, if required by `instantUpdate`
    */
   const handleCheckboxListUpdate = useCallback(
     (newSelection: SelectListProps<T>['value']) => {
+      // only allow updates while open
+      // seems obvious, but animated transitions blur the lines
+      if (!isOpen) return;
```

Comment:
Here we stop any selection updates being made while the popover is closed. This is possible during animations or when the ghost popover component remains active (but not visible) along with the persistent backdrop.